ui/src/hooks/useKubernetes.ts

```diff
@@ -601,18 +601,18 @@ export const initElEvent = (
         text = await readText();
       } catch (e) {
         if (terminalStore.termSelectionText !== '') text = terminalStore.termSelectionText;
+      } finally {
+        socket.send(
+          JSON.stringify({
+            id: kubernetesStore.globalTerminalId,
+            k8s_id: terminalStore.currentTab,
+            type: 'TERMINAL_K8S_DATA',
+            data: text
+          })
+        );
       }
 
       e.preventDefault();
-
-      socket.send(
-        JSON.stringify({
-          id: kubernetesStore.globalTerminalId,
-          k8s_id: terminalStore.currentTab,
-          type: 'TERMINAL_K8S_DATA',
-          data: text
-        })
-      );
     },
     false
   );
```

Comment:
The code does not seem to contain any significant irregularities, potential issues, or optimizations at present. It appears well-written and ready for use.

Here is an update that could help improve readability:

```HTML
export const initElEvent = async (text, textToRead) => { ...
}

const readTextFromLocal = () =>
  new Promise((resolve) => {
    // implementation goes here
  });

```
This way, both lines should be easier to understand, especially `readTextFromLocal` which can simplify future modifications more simply.

Let me know if you need further assistance, and please let me know if you have more questions about this or anything else related to programming.